stl/inc/chrono

```diff
@@ -6202,6 +6163,128 @@ namespace chrono {
+                    }
+                    break;
+
+                case 'b':
+                case 'B':
+                case 'h':
+                case 'm':
+                    if constexpr (is_same_v<_Ty, month>) {
+                        return _Val.ok();
+                    } else if constexpr (_Is_any_of_v<_Ty, month_day, month_day_last, month_weekday, month_weekday_last,
+                                             year_month> || _Is_ymd) {
+                        return _Val.month().ok();
+                    }
+                    break;
+
+                case 'C':
+                case 'y':
+                case 'Y':
+                    if constexpr (is_same_v<_Ty, year>) {
+                        return _Val.ok();
+                    } else if constexpr (_Is_any_of_v<_Ty, year_month> || _Is_ymd) {
+                        return _Val.year().ok();
+                    }
+                    break;
+
+                case 'd':
+                case 'e':
+                    if constexpr (_Is_any_of_v<_Ty, day, month_day_last>) {
+                        return _Val.ok();
+                    } else if constexpr (is_same_v<_Ty, month_day>) {
+                        return _Val.day().ok();
+                    } else if constexpr (_Is_ymd) {
+                        const year_month_day& _Ymd{_Val};
+                        return _Ymd.day().ok();
+                    }
+                    break;
+
+                case 'D':
+                case 'F':
+                    if constexpr (_Has_ok<_Ty>) {
+                        return _Val.ok();
+                    }
+                    break;
+
+                case 'j':
+                    if constexpr (is_same_v<_Ty, month_day>) {
+                        if (_Val.month() > February) {
+                            _THROW(format_error("The day of year for a month_day past February is ambiguous."));
+                        }
+                        return true;
+                    } else if constexpr (is_same_v<_Ty, month_day_last>) {
+                        if (_Val.month() >= February) {
+                            _THROW(
+                                format_error("The day of year for a month_day_last other than January is ambiguous"));
+                        }
+                        return true;
+                    } else if constexpr (_Is_ymd) {
+                        return _Val.ok();
+                    }
+                    break;
+
+                case 'g':
+                case 'G':
+                case 'U':
+                case 'V':
+                case 'W':
+                    if constexpr (_Is_ymd) {
+                        return _Val.ok();
+                    }
+
+                default:
+                    if constexpr (_Has_ok<_Ty>) {
+                        return _Val.ok();
+                    }
+                    return true;
+                }
+                _STL_INTERNAL_CHECK(false);

```

Comment:
The idea is that if a specifier is being checked manually, we should check it for every type that is valid for that specifier (apart from the always correct types at the top). This way, if we miss specific handling for a type-specifier combo, we can catch it internally.